CMakeLists.txt

```diff
@@ -0,0 +1,919 @@
+cmake_minimum_required( VERSION 3.20 )
+cmake_policy( SET CMP0118 NEW )
+
+enable_language( C )
+enable_language( CXX )
+enable_language( Fortran )
+
+project( WRF )
+set( EXPORT_NAME ${PROJECT_NAME} )
+
+if ( DEFINED CMAKE_TOOLCHAIN_FILE )
+  set( WRF_CONFIG ${CMAKE_TOOLCHAIN_FILE} )
+  # message( STATUS "Loading configuration file... : ${WRF_CONFIG}" )
+  # include( ${WRF_CONFIG} )
+endif()
+
+# list( APPEND CMAKE_MODULE_PATH         )
+list( APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/ ${PROJECT_SOURCE_DIR}/cmake/modules )
+
+# Use link paths as rpaths 
+set( CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE )
+set( CMAKE_Fortran_PREPROCESS          ON )
+
+# This is always set
+list( APPEND CMAKE_C_PREPROCESSOR_FLAGS -P -nostdinc -traditional )
+
+include( CMakePackageConfigHelpers )
+include( CheckIPOSupported )
+include( c_preproc   )
+include( m4_preproc  )
+include( target_copy )
+include( confcheck   )
+include( gitinfo     )
+include( printOption )
+include( wrf_case_setup )
+include( wrf_get_version )
+
+check_ipo_supported( RESULT IPO_SUPPORT )
+
+# First grab git info
+wrf_git_commit(
+                RESULT_VAR        GIT_VERSION
+                WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+                )
+
+# Configure file for usage
+configure_file( 
+                ${PROJECT_SOURCE_DIR}/cmake/template/commit_decl.cmake
+                ${PROJECT_BINARY_DIR}/inc/commit_decl
+                @ONLY
+                )
+
+# Grab version info
+wrf_get_version( ${PROJECT_SOURCE_DIR}/README )
+
+################################################################################
+##
+## Options that can be user configured
+##
+################################################################################
+# Mode configuration
+
+set( OPTIMIZATION_LEVEL "" )
+set( WRF_OS             "" )
+set( WRF_MACH           "" )
+
+if ( "${CMAKE_BUILD_TYPE}" STREQUAL "" )
+  set( CMAKE_BUILD_TYPE Debug )
```

Comment:
If my understanding is correct, and the default build type is `Debug`, I would propose we make the default `Release` (or something with optimizations enabled).